scapy/fields.py

```diff
@@ -380,6 +380,8 @@ def __init__(self, name, default):
 class StrField(Field):
     __slots__ = ["remain"]
     def __init__(self, name, default, fmt="H", remain=0):
+        if isinstance(default, str) and default == "":
```

Comment:
I don't get it. Why would you make a special case for the empty string? I think we should fix all the default values (or add this hack for any `str` values, but I prefer the former).

Reply:
Hmmm, not sure why i did that....